src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -147,15 +157,27 @@ protected override void ApplyResponseChallenge()
                 properties = new AuthenticationProperties(ChallengeContext.Properties);
             }
 
-            if (string.IsNullOrEmpty(properties.RedirectUri))
+            string redirectToUse = null;
```

Comment:
Hum, all this stuff seems really odd: remember that `properties.RedirectUri` corresponds to the final location the user agent will be redirect to after the whole authentication/authorization process, not the `redirect_uri` (as defined by the OIDC specs) used for the callback part.